src/document/json/tree.ts

```diff
@@ -339,40 +375,121 @@ export class Tree {
     if (fromIdx > toIdx) {
       throw new Error('from should be less than or equal to to');
     }
-
-    const crdtNodes: Array<CRDTTreeNode> = contents
-      .map((content) => content && createCRDTTreeNode(this.context!, content))
-      .filter((a) => a) as Array<CRDTTreeNode>;
-    const fromPos = this.tree.findPos(fromIdx);
-    const toPos = this.tree.findPos(toIdx);
-    const ticket = this.context.getLastTimeTicket();
-    this.tree.edit(
-      [fromPos, toPos],
-      crdtNodes.length
-        ? crdtNodes.map((crdtNode) => crdtNode?.deepcopy())
-        : undefined,
-      ticket,
-    );
-
-    this.context.push(
-      TreeEditOperation.create(
-        this.tree.getCreatedAt(),
-        fromPos,
-        toPos,
-        crdtNodes.length ? crdtNodes : undefined,
+    if (contents.length !== 0 && contents[0] !== null) {
```

Comment:
what if `contents[0]` is `undefined`?

I think it's better to use `!contents[0]` rather than `contents[0] !== null` which can filter both `undefined` and `null`